src/main/kotlin/apply/ui/admin/evaluation/EvaluationForm.kt

```diff
@@ -0,0 +1,67 @@
+package apply.ui.admin.evaluation
+
+import apply.application.EvaluationRequest
+import apply.domain.evaluation.Evaluation
+import apply.domain.recruitment.Recruitment
+import com.vaadin.flow.component.button.Button
+import com.vaadin.flow.component.textfield.TextArea
+import com.vaadin.flow.component.textfield.TextField
+import support.views.BindingFormLayout
+import support.views.createErrorSmallButton
+import support.views.createItemSelect
+import support.views.createPrimarySmallButton
+
+class EvaluationForm() : BindingFormLayout<EvaluationRequest>(EvaluationRequest::class) {
+    private val title: TextField = TextField("평가명")
+    private val description: TextArea = TextArea("설명")
+    private val recruitment = createItemSelect<Recruitment>("모집").apply {
+        setItemLabelGenerator(Recruitment::title)
+        isEmptySelectionAllowed = false
+    }
+    private val beforeEvaluation = createItemSelect<Evaluation>("이전 평가").apply {
+        setItemLabelGenerator(Evaluation::title)
+    }
+    private val evaluationItems: MutableList<EvaluationItemForm> = mutableListOf()
+
+    constructor(recruitments: List<Recruitment>, listener: (id: Long) -> List<Evaluation>) : this() {
+        recruitment.setItems(recruitments)
+        recruitment.addValueChangeListener {
+            beforeEvaluation.setItems(listener.invoke(it.value.id))
```

Comment:
`invoke`는 생략할 수 있어요.

```suggestion
            beforeEvaluation.setItems(listener(it.value.id))
```

Reply:
아직까지 코틀린을 모르는게 많네요!
수정했습니다.